### ISSUE TITLE
[market api] RejectProposal seems to complete successfully but returns HTTP 500?

### ISSUE THREAD

**Issue by stranger80** [golemfactory/yagna#987]
```
yagna -V
yagna 0.6.0-rc2 (d3219203 2021-01-27)
```

Hi guys, can you please help me troubleshoot one of my test scenarios?

The scenario is a typical Requestor sequence: SubscribeDemand, Collect, and then Reject on received proposals.

Collect:

```
GET http://127.0.0.1:7465/market-api/v1/demands/78dc8e2cbb864957877a38ab36113aa2-d397967f540038c74594a019a6c781c3cd4734647bdfdefcee53aadc4b142e54/events?timeout=30 HTTP/1.1
Host: 127.0.0.1:7465
Authorization: Bearer 90bdf868625f44608cd766230ffda387
Accept: application/json
User-Agent: OpenAPI-Generator/1.0.0/csharp
Connection: Keep-Alive
Accept-Encoding: gzip, deflate
```

returns:

```
HTTP/1.1 200 OK
content-length: 3246
content-type: application/json
date: Thu, 28 Jan 2021 20:39:58 GMT

[{"eventType":"ProposalEvent","eventDate":"2021-01-28T20:39:58.862Z","proposal":{"properties":{"golem.activity.caps.transfer.protocol":["http","https","gftp"],"golem.com.payment.debit-notes.acceptance-timeout":120,"golem.com.payment.platform.erc20-rinkeby-tglm.address":"0x9c72c27a687134299037e464912c90af120d8dc1","golem.com.payment.platform.zksync-rinkeby-tglm.address":"0x9c72c27a687134299037e464912c90af120d8dc1","golem.com.pricing.model":"linear","golem.com.pricing.model.linear.coeffs":[0.002,0.001,0.0],"golem.com.scheme":"payu","golem.com.scheme.payu.interval_sec":6.0,"golem.com.usage.vector":["golem.usage.duration_sec","golem.usage.cpu_sec"],"golem.inf.cpu.architecture":"x86_64","golem.inf.cpu.cores":4,"golem.inf.cpu.threads":2,"golem.inf.mem.gib":4.0,"golem.inf.storage.gib":98.58828353881836,"golem.node.debug.subnet":"devnet-alpha.4","golem.node.id.name":"mf.4","golem.runtime.name":"wasmtime","golem.runtime.version":"0.2.1","golem.srv.caps.multi-activity":true},"constraints":"(&\n  (golem.srv.comp.expiration>1611822663872)\n  (golem.node.debug.subnet=devnet-alpha.4)\n)","proposalId":"R-63aa954b0016b26a2baf1087d81772239f6a78dcff1fe5a2457d90ef1be11c67","issuerId":"0x9c72c27a687134299037e464912c90af120d8dc1","state":"Initial","timestamp":"2021-01-28T20:39:58.859534500Z"}}
]
```

Then a RejectProposal call:

```
POST http://127.0.0.1:7465/market-api/v1/demands/78dc8e2cbb864957877a38ab36113aa2-d397967f540038c74594a019a6c781c3cd4734647bdfdefcee53aadc4b142e54/proposals/R-63aa954b0016b26a2baf1087d81772239f6a78dcff1fe5a2457d90ef1be11c67/reject HTTP/1.1
Host: 127.0.0.1:7465
Authorization: Bearer 90bdf868625f44608cd766230ffda387
Accept: application/json
User-Agent: OpenAPI-Generator/1.0.0/csharp
Connection: Keep-Alive
Accept-Encoding: gzip, deflate
Content-Type: application/json
Content-Length: 19

{"message":"dblah"}
```

returns:

```
HTTP/1.1 500 Internal Server Error
content-length: 124
content-type: application/json
date: Thu, 28 Jan 2021 20:39:58 GMT

{"message":"Proposal [P-63aa954b0016b26a2baf1087d81772239f6a78dcff1fe5a2457d90ef1be11c67] not found (subscription [None])."}
```

The logs include:

```
[2021-01-28T20:39:58Z INFO  ya_market::negotiation::common] Requestor [0x0c4a4a31a2244ff70547afdefc6497879a34cab4] rejected Proposal [R-63aa954b0016b26a2baf1087d81772239f6a78dcff1fe5a2457d90ef1be11c67] with reason: 'dblah'.
[2021-01-28T20:39:59Z INFO  actix_web::middleware::logger] 127.0.0.1:4272 "POST /market-api/v1/demands/78dc8e2cbb864957877a38ab36113aa2-d397967f540038c74594a019a6c781c3cd4734647bdfdefcee53aadc4b142e54/proposals/R-63aa954b0016b26a2baf1087d81772239f6a78dcff1fe5a2457d90ef1be11c67/reject HTTP/1.1" 500 124 "-" "OpenAPI-Generator/1.0.0/csharp" 0.113432
```

...so really no signs of Internal Server Error, etc.

Is there an issue, or is stg wrong with the RejectProposal request?

Help appreciated.

**Comment by tworec** [comment]
@stranger80 pls specify `yagna -V` which you are using

**Comment by tworec** [comment]
closed via #988
